### ISSUE TITLE
Document and validate Table Transformer ONNX output contract expectations

### ISSUE THREAD

**Issue by lfgranja** [HKUDS/RAG-Anything#205]
## Background

The `TableProcessor` in `src/table_transformer.rs` makes specific assumptions about ONNX model output format. These assumptions are not documented, making it difficult to:
1. Know which ONNX model exports are compatible
2. Debug inference failures
3. Support alternative model variants

## Current Implementation

The current code at `src/table_transformer.rs:139-164` assumes:

### Output Tensor Name
- Primary: `"output0"`
- Fallback: `"output"`

### Output Shape & Format
- **Flat array format**: `len % 6 == 0`
- **Per-detection fields** (6 values each):
  ```
  [x1, y1, x2, y2, confidence, class_id]
  ```
- **Coordinate format**: Absolute or normalized pixel coordinates (auto-detected via threshold `<= 1.5`)

### Class Mapping
| Class ID | Label |
|----------|-------|
| 0 | `table` |
| 1 | `table_row` |
| 2 | `table_column` |
| 3 | `table_cell` |

### 1. Microsoft Table Transformer (Hugging Face Optimum Export)

**Source**: `microsoft/table-transformer-structure-recognition` exported via `optimum.exporters.onnx`

**Output Tensors**:
- `logits`: Shape `(batch_size, num_queries, num_classes + 1)` - Classification scores including no-object class
- `pred_boxes`: Shape `(batch_size, num_queries, 4)` - Normalized boxes in `(center_x, center_y, width, height)` format

**Post-processing Required**:
- Apply softmax to logits
- Filter by confidence threshold
- Convert from `(cx, cy, w, h)` to `(x1, y1, x2, y2)` format
- Scale to original image dimensions

Reference: [Hugging Face Table Transformer Documentation](https://huggingface.co/docs/transformers/model_doc/table-transformer)

### 2. YOLO-style Export (Ultralytics)

**Output Format**:
- Single tensor: `(batch, num_detections, 6)` or `(batch, 6, num_detections)`
- Fields: `[x1, y1, x2, y2, confidence, class_id]`

**This is the format currently expected by the code.**

### 3. DETR-style Export (Raw)

**Output Tensors**:
- Multiple separate outputs for bbox predictions and class logits
- Requires custom post-processing similar to Hugging Face

## Problems with Current Implementation

1. **Incompatible with Hugging Face exports**: The standard `microsoft/table-transformer-*` ONNX exports use separate `logits` and `pred_boxes` tensors, not a flat 6-field array.

2. **No shape validation**: Code only checks `len % 6 == 0` but doesn't validate tensor dimensions.

3. **No tensor name documentation**: Users don't know which tensor names are expected.

4. **Coordinate format ambiguity**: The auto-detection of normalized vs absolute coordinates (`x2 <= 1.5`) is fragile.

5. **No model provenance tracking**: No way to know which model variant was used during inference.

### Phase 1: Documentation (This Issue)

Add comprehensive documentation to `table_transformer.rs`:

```rust
/// Table Transformer ONNX Model Requirements
/// ==========================================
/// 
/// This processor expects ONNX models with the following output contract:
///
/// ## Supported Output Formats
///
/// ### Format A: Flattened Detection Array (Currently Supported)
/// - **Tensor name**: \`output0\` or \`output\`
/// - **Shape**: \`(N * 6,)\` where N is number of detections
/// - **Fields per detection**: \`[x1, y1, x2, y2, confidence, class_id]\`
/// - **Coordinate format**: Pixel coordinates (absolute) or normalized [0,1]
/// - **Class IDs**: 0=table, 1=row, 2=column, 3=cell
///
/// ### Known Compatible Models
/// - Custom YOLO-style table structure exports
/// - Some third-party table-transformer ONNX conversions
///
/// ### Incompatible Models (Need Conversion)
/// - \`microsoft/table-transformer-structure-recognition\` (Hugging Face)
/// - \`microsoft/table-transformer-detection\` (Hugging Face)
/// - Raw DETR-style exports with separate logits/boxes outputs
```

### Phase 2: Enhanced Validation

Add runtime validation with clear error messages:

```rust
fn validate_onnx_output(outputs: &HashMap<String, Value>) -> Result<OutputFormat> {
    // Check for known output names
    // Validate tensor shapes
    // Return detected format or error with guidance
}
```

### Phase 3: Multi-Format Support (Future)

Support multiple ONNX output formats:
- Hugging Face standard export (logits + pred_boxes)
- YOLO-style flattened format
- Configurable via `TableModelConfig`

### DETR Standard Output Format

From Facebook DETR models:
```python
output_names = ["pred_logits", "pred_boxes"]
```

### Output Tensor Specifications

**pred_logits (Classification Logits)**:
- **Purpose**: Class probability scores for each predicted object
- **Shape**: `[batch, num_queries, num_classes]` 
- **Typical values**:
  - Table Structure: `[1, 100, 7]` (background + 6 table element types)
  - Table Detection: `[1, 100, 2]` (background + table)
- **Data type**: `float32`

**pred_boxes (Bounding Boxes)**:
- **Purpose**: Normalized bounding box coordinates
- **Shape**: `[batch, num_queries, 4]`
- **Coordinate format**: `[cx, cy, w, h]` (center x, center y, width, height)
- **Normalization**: Boxes are typically output in [0, 1] range

### Post-processing Requirements

**DETR does NOT require NMS (Non-Maximum Suppression)** - Unlike YOLO/Faster-RCNN, DETR uses a bipartite matching loss that produces a fixed number of predictions.

**Score Calculation**:
```python
# Get class probabilities from logits
probabilities = pred_logits.softmax(-1)

# Keep predictions with confidence above threshold
keep = probabilities.max(-1).values > confidence_threshold
```

**Box Coordinate Transformation**:
```python
def rescale_bboxes(boxes, original_image_size):
    """Convert normalized [cx, cy, w, h] boxes to absolute [x1, y1, x2, y2] coordinates"""
    cx = boxes[:, 0] * original_image_size[1]
    cy = boxes[:, 1] * original_image_size[0]
    
    x1 = cx - (boxes[:, 2] / 2) * original_image_size[1]
    y1 = cy - (boxes[:, 3] / 2) * original_image_size[0]
    x2 = cx + (boxes[:, 2] / 2) * original_image_size[1]
    y2 = cy + (boxes[:, 3] / 2) * original_image_size[0]
    
    return np.stack([x1, y1, x2, y2], axis=1)
```

## Known Issues from Research

1. **pixel_mask Input Shape Mismatch** ([Issue #120](https://github.com/microsoft/table-transformer/issues/120)):
   - The `pixel_mask` input becomes `[1, 64, 64]` in ONNX, which differs from the original PyTorch model

2. **Deformable Attention Not Supported**:
   - Deformable DETR variants cannot be exported to ONNX directly
   - Need to use standard DETR or Table Transformer variants

## Resources

- [Microsoft Table Transformer Paper](https://arxiv.org/abs/2110.00061)
- [Hugging Face Model Documentation](https://huggingface.co/docs/transformers/model_doc/table-transformer)
- [PubTables-1M Dataset](https://github.com/microsoft/table-transformer)
- [ONNX Export Tutorial for DETR models](https://github.com/microsoft/table-transformer/issues/56)
- [Xenova ONNX Exports](https://huggingface.co/Xenova/table-transformer-structure-recognition)

## Acceptance Criteria

- [ ] Add module-level documentation explaining ONNX requirements
- [ ] Document expected tensor names, shapes, and field layouts
- [ ] Add inline comments in `detect_elements()` and `postprocess_detections()`
- [ ] Create `MODELS.md` or section in README with compatible model list
- [ ] Add validation logging that reports detected format at startup
- [ ] Update error messages to guide users when incompatible models are used

## Related

- Issue #204 (imageproc edge detection refactor) - Alternative fallback
- Review feedback from PR #9 (qodo-code-review, gemini-code-assist)
